lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -494,7 +494,8 @@ private ColumnChunkMetadata getColumnChunkMetaData(BlockMetadata blockMetaData,
             throws IOException
     {
         for (ColumnChunkMetadata metadata : blockMetaData.getColumns()) {
-            if (metadata.getPath().equals(ColumnPath.get(columnDescriptor.getPath()))) {
+            // Column paths for nested structures have common root, so we compare in reverse to find mismatch sooner
+            if (arrayEqualsReversed(metadata.getPath().toArray(), columnDescriptor.getPath())) {
```

Comment:
`ColumnPath.toArray` **should** copy

Reply:
True, but I think there should also be a `boolean equals(String[] path)` method to allow equality check without needing array copy.
I think we need to switch to our version of that class eventually, it just needs moving some more parquet-mr code into Trino.